src/lsdb/catalog/dataset/healpix_dataset.py

```diff
@@ -76,3 +89,20 @@ def get_partition_index(self, order: int, pixel: int) -> int:
             raise ValueError(f"Pixel at order {order} pixel {pixel} not in Catalog")
         partition_index = self._ddf_pixel_map[hp_pixel]
         return partition_index
+
+    def query(self, expr: str) -> Self:
+        """Filters catalog using a complex query expression
+
+        Args:
+            expr (str): Query expression to evaluate. The column names that are not valid Python
+                variables names should be wrapped in backticks, and any variable values can be
+                injected using f-strings. The use of '@' to reference variables is not supported.
+                More information about pandas query strings is available
+                `here <https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.query.html>`__.
+
+        Returns:
+            A catalog that contains the data from the original catalog that complies
+            with the query expression
+        """
+        ddf = self._ddf.query(expr)
+        return self.__class__(ddf, self._ddf_pixel_map, self.hc_structure)
```

Comment:
This is a great idea! We should probably do the same for `assign` and `head`.

Reply:
Make an issue #178